index.js

```diff
@@ -686,6 +686,7 @@ RedisClient.prototype.return_reply = function (reply) {
             throw new Error("subscriptions are active but got an invalid reply: " + reply);
         }
     } else if (this.monitoring) {
+        reply = reply && reply.toString && reply.toString();
         len = reply.indexOf(" ");
```

Comment:
That's why I added the coercion to string if the return type is buffer :) But I'm still uncertain if that's the right thing to do, as the request was to only return buffers instead of strings.